nncf/config/extractors.py

```diff
@@ -11,64 +11,49 @@
  limitations under the License.
 """
 
-from copy import deepcopy
-from typing import Dict, List, Optional
+from typing import Dict
+from typing import List
+from typing import Optional
 
-from nncf.common.hardware.config import HW_CONFIG_TYPE_TARGET_DEVICE_MAP
-from nncf.common.hardware.config import HWConfigType
 from nncf.common.quantization.initialization.range import PerLayerRangeInitConfig
 from nncf.common.quantization.initialization.range import RangeInitConfig
 from nncf.common.utils.logger import logger
-from nncf.config import NNCFConfig
+from nncf.config.config import NNCFConfig
 from nncf.config.structures import BNAdaptationInitArgs
 from nncf.config.structures import QuantizationRangeInitArgs
 
 
-def extract_compression_algorithm_configs(config: NNCFConfig) -> List[NNCFConfig]:
-    """
-    Extracts specific algorithm parameters for each compression algorithm from the
-    common NNCFConfig.
+def extract_algorithm_names(config: NNCFConfig) -> List[str]:
+    retval = []
+    compression_config_json_section = config.get('compression', [])
+    if isinstance(compression_config_json_section, dict):
+        compression_config_json_section = [compression_config_json_section]
+    for algo_config in compression_config_json_section:
+        retval.append(algo_config['algorithm'])
+    return retval
 
-    :param config: An instance of the NNCFConfig.
-    :return: List of the NNCFConfigs, each of which contains specific algorithm
-        parameters related only to this algorithm
-    """
-    compression_config_json_section = config.get('compression', {})
-    compression_config_json_section = deepcopy(compression_config_json_section)
 
-    hw_config_type = None
-    target_device = config.get('target_device', 'ANY')
-    if target_device != 'TRIAL':
-        hw_config_type = HWConfigType.from_str(HW_CONFIG_TYPE_TARGET_DEVICE_MAP[target_device])
-    global_compression_lr_multiplier = config.get('compression_lr_multiplier', None)
-    global_ignored_scopes = config.get("ignored_scopes")
-    global_target_scopes = config.get("target_scopes")
+def extract_algo_specific_config(config: NNCFConfig, algo_name_to_match: str) -> Dict:
```

Comment:
Why you removed docstring of this function?

Reply:
The function changed. I added a proper docstring.